engine.py

```diff
@@ -113,7 +113,7 @@ def post_app_init(self):
         
         # store the menu cmds so that we can access them later from C#
         self._menu_cmd_items = {}
-        for (cmd_name, cmd_details) in self.commands.items():
+        for (cmd_name, cmd_details) in list(self.commands.items()):
```

Comment:
why can't you iterate over the items?

Reply:
same question with the other two times you make a list below

Reply:
Sorry about that, I used 2to3 to help me, it gave me warning messages about `items` not returning a list, but yes it's still iterable, and I guess better memory and time wise.